presto-main/src/main/java/com/facebook/presto/sql/planner/LiteralInterpreter.java

```diff
@@ -133,13 +134,12 @@ else if (value.equals(Double.POSITIVE_INFINITY)) {
             }
         }
 
-        if (type.equals(VARCHAR)) {
+        if (type instanceof VarcharType) {
             if (object instanceof Slice) {
-                return new StringLiteral(((Slice) object).toStringUtf8());
+                return new Cast(new StringLiteral(((Slice) object).toStringUtf8()), type.getTypeSignature().toString(), false, true);
```

Comment:
Why do we need this cast? It's introducing a seemingly unnecessary cast in queries like:

```
presto:tiny> explain select 'abc' from orders;
                                    Query Plan                                     
-----------------------------------------------------------------------------------
 - Output[_col0] => [expr:varchar(3)]                                              
         _col0 := expr                                                             
     - Exchange[GATHER] => expr:varchar(3)                                         
         - Project => [expr:varchar(3)]                                            
                 expr := CAST('abc' AS VARCHAR(3))                                 
             - TableScan[tpch:tpch:orders:sf0.01, originalConstraint = true] => [] 
```